src/modules/feature-modules/innovator/guards/check-status.guard.ts

```diff
@@ -0,0 +1,35 @@
+import { inject } from '@angular/core';
+import { AuthenticationStore, ContextService } from '@modules/stores';
+import { ActivatedRouteSnapshot, CanActivateFn, Router } from '@angular/router';
+import { InnovationStatusEnum } from '@modules/stores/innovation';
+import { catchError, map } from 'rxjs/operators';
+import { of } from 'rxjs';
+
+export function checkStatusGuard(statusList: InnovationStatusEnum[], blockList: boolean = false): CanActivateFn {
+  return (route: ActivatedRouteSnapshot) => {
+    const router: Router = inject(Router);
+    const contextService: ContextService = inject(ContextService);
+    const authenticationStore: AuthenticationStore = inject(AuthenticationStore);
+
+    return contextService
+      .getInnovationContextInfo(route.params.innovationId, authenticationStore.getUserContextInfo())
```

Comment:
I think you should be using the `getOrLoadInnovation` from context store here

Reply:
Right, correct. Changed it as suggested